main_push.go

```diff
@@ -19,11 +19,23 @@ func processGitHubPush(
 	repoOrg := push.GetRepo().GetOrganization()
 	refName := push.GetRef()
 
-	log.Debugf("Got push event :: repo %s :: ref %s", repoName, refName)
+	if len(conf.reposSyncList) == 0 || isRepoManaged(repoName, conf.reposSyncList) {
+		log.Debugf("Got push event :: repo %s :: ref %s", repoName, refName)
 
-	err := syncRemoteRef(log, repoOrg, repoName, refName, conf)
-	if err != nil {
-		log.Errorf("Could not sync branch: %s", err.Error())
+		err := syncRemoteRef(log, repoOrg, repoName, refName, conf)
+		if err != nil {
+			log.Errorf("Could not sync branch: %s", err.Error())
+			return err
+		}
 	}
```

Comment:
I would add here also another debug log entry, like "not syncing repository ______".

Or, even more accurate, restore the `log.Debugf("Got push event :: repo %s :: ref %s", repoName, refName)` to where it was (so that it is printed every time there is a push event, as the message suggests), and here only another message like "repo ______ not being watched, ignoring"